File: solutions/009/_009.java
public class _009 {
    public static void main(String[] args) {
        int s = 1000;
        int r = (int) Math.sqrt(s);
        for (int m = r; m > 1; m--)
            for (int n = m; n > 2; n--) {
                int a = m * m - n * n;
                int b = 2 * (m * n);
                int c = m * m + n * n;
                if (a + b + c == s) {
                    System.out.println(a * b * c);
                    return;
                }
            }
    }
}

File: solutions/009/009.go
package main

import (
	"fmt"
	"math"
)

func main() {
	s := 1000
	r := int(math.Sqrt(float64(s)))
	for m := r; m > 1; m-- {
		for n := m; n > 2; n-- {
			a := m*m - n*n
			b := 2 * (m * n)
			c := m*m + n*n
			if a+b+c == s {
				fmt.Printf("%d\n", a*b*c)
				return
			}
		}
	}
}
